packages/core/src/app/payment/Payment.tsx

```diff
@@ -372,24 +372,25 @@ class Payment extends Component<
 
             if (isErrorWithType(error)) {
                 const { body, headers, status } = error;
-
-                if (body.type === 'provider_error' && headers.location) {
-                    window.top?.location.assign(headers.location);
-                }
-
-                // Reload the checkout object to get the latest `shouldExecuteSpamCheck` value,
-                // which will in turn make `SpamProtectionField` visible again.
-                // NOTE: As a temporary fix, we're checking the status code instead of the error
-                // type because of an issue with Nginx config, which causes the server to return
-                // HTML page instead of JSON response when there is a 429 error.
-                if (
-                    status === 429 ||
-                    body.type === 'spam_protection_expired' ||
-                    body.type === 'spam_protection_failed'
-                ) {
-                    this.setState({ didExceedSpamLimit: true });
-
-                    await loadCheckout();
+                if (body || status === 429){
```

Comment:
As a recent bugfix involved checking `error.body` on line 373, we might not need to recheck on line 375.
https://github.com/bigcommerce/checkout-js/pull/1057/files

Reply:
Nice catch. Thanks for taking a look at this. I've tested this out again with the latest checkout-js and the error can be cleared with that change.  I'll keep an eye out for the bump in BCApp and close [CHECKOUT-7048](https://bigcommercecloud.atlassian.net/browse/CHECKOUT-7048) accordingly.

Reply:
Thanks @bc-peng and @huntario